Adding new option

cs-tushar-kanade opened the pull request:
New option to filter data set based on regex and 'isin' filter

https://pmdb.fortinet.com/ProjectManagement/editprojects.php?id=21237

cs-tushar-kanade committed:
Adding new option

New option to filter data set based on regex and 'isin' filter

https://pmdb.fortinet.com/ProjectManagement/editprojects.php?id=21237

cs-tushar-kanade committed:
typo correction

typo correction

cs-tushar-kanade committed:
correcting variable name

correcting variable name

cs-pooja-singh commented on csv-data-management/info.json:
```diff
@@ -1,6 +1,6 @@
-  "version": "1.0.0",
+  "version": "1.1.0",
   "label": "CSV Data Management",
   "description": "CSV Data management can perform different operations on CSV files like read file, perform deduplication, merge two CSV files,join two CSV files,concat two CSV files and return well formatted dataset",
```
files, join two CSV files,concat two CSV files, and return formatted dataset
(spaces after commas)

cs-tushar-kanade replied:
Done

cs-pooja-singh commented on csv-data-management/extract_data_from_csv.py:
```diff
@@ -395,5 +412,30 @@ def handle_params(params,file_param):
                 raise ConnectorError('Invalid File IRI {0}'.format(value))
```
this will be caught again in line 413 and raised as /api/3/attachments not found. Do you want to handle the two separately?

cs-tushar-kanade replied:
handle_params , I picked this function from VT connector, I will check if i can optimize  it

cs-tushar-kanade replied:
Will work on optimization in next version

cs-pooja-singh commented on csv-data-management/extract_data_from_csv.py:
```diff
@@ -395,5 +412,30 @@ def handle_params(params,file_param):
+
+    # Filter dataset 
+    if regex:
+        reg = params.get('filter')
```
why not set these directly in 422 instead of initializing regex?

cs-tushar-kanade replied:
did changes , will update PR

cs-pooja-singh commented on csv-data-management/extract_data_from_csv.py:
```diff
@@ -395,5 +412,30 @@ def handle_params(params,file_param):
+    return df
+
+def _df_to_csv(df):
+    df.to_csv('/tmp/dataset.csv', encoding='utf-8', header='true',compression="zip")
```
do you want to append a unique uuid to the name? will override each other if running in parallel playbooks.
also, when is this file cleaned up?

cs-tushar-kanade replied:
did changes , will update PR

cs-tushar-kanade replied:
Done

cs-pooja-singh commented on csv-data-management/info.json:
```diff
@@ -75,6 +75,61 @@
+          "editable": true,
+          "filtervalue": "",
+          "options": [
+            "Use Regex as Filter",
```
"On specified values", "On values matching a regex"

cs-pooja-singh commented on csv-data-management/info.json:
```diff
@@ -75,6 +75,61 @@
+
+            },
+            { 
+              "title": "Regex Filter",
```
can we add a placeholder with a sample regex?

cs-tushar-kanade replied:
done , added placeholder regex for Ip address

cs-pooja-singh commented on csv-data-management/info.json:
```diff
@@ -75,6 +75,61 @@
+
+            },
+              {
+              "title": "Filter String",
```
can we add a tooltip that says that user needs to specify a comma separated list of strings?
and a placeholder with an example

cs-tushar-kanade replied:
done

cs-tushar-kanade committed:
PR feedback

incorporating PR feedback.
Still working on rest of items

cs-tushar-kanade committed:
updating connector name 

updating connector name  in json

cs-tushar-kanade committed:
Adding option to save dataset

Adding option to save dataset as  attachment record  with CSV file .
To support this option , have updated output schema.
Solution pack will be required to be updated to support change

cs-tushar-kanade committed:
Updated info json and attachment creation option

Updated info json and attachment creation option

cs-tushar-kanade committed:
Updating JSON

Updating JSON

cs-tushar-kanade committed:
PR Feedback

PR Feedback

cs-tushar-kanade committed:
Punctuation Correction

Punctuation Correction

cs-tushar-kanade committed:
optimising code

optimising code

cs-tushar-kanade committed:
Code optimisation

Code optimisation
